Gems/SurfaceData/Code/CMakeLists.txt

```diff
@@ -97,9 +98,11 @@ if(PAL_TRAIT_BUILD_TESTS_SUPPORTED)
         BUILD_DEPENDENCIES
             PRIVATE
                 AZ::AzTest
+                AZ::AzToolsFramework
                 Legacy::CryCommon
                 Gem::SurfaceData.Static
                 Gem::LmbrCentral
+                Gem::Terrain
```

Comment:
Terrain definitely shouldn't be added.  Why is AzToolsFramework needed for the tests?  That's a surprising addition.